### PR TITLE
Fix google login

### PR DESCRIPTION
**Pull request by DaniilVolobuiev**
## Describe your changes

> GetMe endpoint.
>Google oauth fix
>Login also returns a user now
>updated patch auth guard to cut bearer prefix in token 

## Issue ticket code (and/or) and link

> [Link to JIRA ticket]()

### General

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -154,11 +141,17 @@ export default class AuthService {
...
     );
 
-    res.cookie('accessToken', accessToken, this.accessTokenCookieOptions);
-    res.redirect(this.configService.get('CLIENT_URL'));
+    res.clearCookie('accessToken');
+    res.cookie('accessToken', accessToken);
+
+    res.redirect(this.configService.get('SECOND_FORM_URL'));
+
+    // res.redirect(this.configService.get('CLIENT_API'));
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: why we need cookie? I dont like this solution as server and frontend will be on different servers, its mean that cookie will not work

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -278,12 +277,40 @@ export default class AuthService {
...
+      const { token } = await this.generateToken(doctor);
+      const userInfo = this.stripPassword(doctor);
+
+      return { token, userInfo };
     } catch (err) {
       throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
+
+  private stripPassword(doctor: Doctor): UserInfo {
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: what "strip" mean?

File: src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -278,12 +277,40 @@ export default class AuthService {
...
+    const { password, activationLink, ...userInfo } = doctor;
+
+    return userInfo;
+  }
+
+  async getMe(req: Request): Promise<UserInfo> {
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts:307]: req is not the name that show what variable means
also its strange that you don't have user data from middleware
this service look very strange

File: src/modules/auth/auth.service.ts (hunk 2 above)
Comment (DaniilVolobuiev) [reply src/modules/auth/auth.service.ts]: Edited the name of function

File: src/modules/auth/auth.service.ts (hunk 3 above)
Comment (DaniilVolobuiev) [reply src/modules/auth/auth.service.ts:307]: Now service uses middleware to find the user by id. Fixed the ts mistake. I also dont understand what do you mean saying this: "req is not the name that show what variable means". It is request from where i take the token.

File: src/modules/auth/auth.controller.ts [hunk 4]
```diff
@@ -8,87 +8,106 @@ import {
...
 import Doctor from '../doctor/entity/doctor.entity';
 import AuthService from './auth.service';
 import LoginDoctorDto from '../doctor/dto/login-doctor.dto';
 
 import CreateDoctorDto from '../doctor/dto/create-doctor.dto';
 import JwtAuthGuard from './utils/Guards';
-import ForgotPasswordDto from "../doctor/dto/forgot-password.dto";
-import ResetPasswordDto from "../doctor/dto/change-password.dto";
+import ForgotPasswordDto from '../doctor/dto/forgot-password.dto';
+import ResetPasswordDto from '../doctor/dto/change-password.dto';
```
Comment (lyudad) [inline src/modules/auth/auth.controller.ts]: why its not absolute path?

Comment (DaniilVolobuiev) [reply src/modules/auth/auth.controller.ts]: it is not my code, but i fixed it)

**Pull request merged**